go/nbs/store.go

```diff
@@ -129,15 +149,70 @@ func (nbs *NomsBlockStore) Put(c chunks.Chunk) {
 	nbs.stats.PutLatency.SampleTime(dur)
 }
 
+type byteTableReader struct {
+	tableReader
+	h    addr
+	data []byte
+}
+
+func (br byteTableReader) ReadAt(p []byte, off int64) (n int, err error) {
+	d.PanicIfTrue(off+int64(len(p)) > int64(len(br.data)))
+	copy(p, br.data[off:off+int64(len(p))])
+	return len(p), nil
+}
+
+func (br byteTableReader) hash() addr {
+	return br.h
+}
+
+// MUST be called already holding the store lock. Makes durable the existing
+// memtable. Will block if there are already |nbs.persistLimit| concurrent
+// persist operations underway until one has completed.
+func (nbs *NomsBlockStore) persistMemTable() {
+	if nbs.mt == nil || nbs.mt.count() == 0 {
+		return // nothing to do
+	}
+
+	mt := nbs.mt
+	nbs.mt = nil
+
+	h, data, chunkCount := mt.write(nbs.tables, nbs.stats)
```

Comment:
One of the design constraints for `persistingChunkSource` was to do the work of serializing a memTable outside of the storewide lock and, ideally, without holding up the `Put()` that put us over the memTable size limit. Does this new approach make that impossible? That seems like a meaningful degradation to me.